spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/trace/StackdriverTraceAutoConfiguration.java

```diff
@@ -121,19 +126,21 @@ public SleuthProperties stackdriverSleuthProperties(SleuthProperties sleuthPrope
 		return sleuthProperties;
 	}
 
-	@Bean
-	@ConditionalOnMissingBean(name = "traceSenderThreadPool")
-	public ThreadPoolTaskScheduler traceSenderThreadPool(GcpTraceProperties traceProperties) {
-		ThreadPoolTaskScheduler scheduler = new ThreadPoolTaskScheduler();
-		scheduler.setPoolSize(traceProperties.getNumExecutorThreads());
-		scheduler.setThreadNamePrefix("gcp-trace-sender");
-		scheduler.setDaemon(true);
-		return scheduler;
-	}
-
 	@Bean
 	@ConditionalOnMissingBean(name = "traceExecutorProvider")
-	public ExecutorProvider traceExecutorProvider(@Qualifier("traceSenderThreadPool") ThreadPoolTaskScheduler scheduler) {
+	public ExecutorProvider traceExecutorProvider(GcpTraceProperties traceProperties, @Qualifier("traceSenderThreadPool") Optional<ThreadPoolTaskScheduler> userProvidedScheduler) {
```

Comment:
`Optional` works here? I thought we have to use `ObjectProvider`.
cc/ @artembilan

Reply:
I think it's OK to stick with `Optional` for simple use-cases.